common/src/tap/checks/value_check.rs

```diff
@@ -0,0 +1,535 @@
+                match deployment.as_str() {
+                    "global" => {
+                        *self.global_model.write().unwrap() = None;
+                    }
+                    deployment_id => match DeploymentId::from_str(deployment_id) {
+                        Ok(deployment_id) => {
+                            self.cost_models.write().unwrap().remove(&deployment_id);
+                        }
+                        Err(_) => {
+                            error!(
+                                "Received delete request for an invalid deployment_id: {}",
+                                deployment_id
+                            )
+                        }
+                    },
+                };
+            }
+            // UPDATE and TRUNCATE are not expected to happen. Reload the entire cost
+            // model cache.
+            Err(_) => {
+                error!(
+                    "Received an unexpected cost model table notification: {}. Reloading entire \
+                                cost model.",
+                    payload
+                );
+
+                MinimumValue::value_check_reload(
+                    &self.pgpool,
+                    self.cost_models.clone(),
+                    self.global_model.clone(),
+                )
+                .await
+                .expect("should be able to reload cost models")
+            }
+        }
+    }
+}
+
+impl Drop for MinimumValue {
+    fn drop(&mut self) {
+        // Clean shutdown for the minimum value check
+        // Though since it's not a critical task, we don't wait for it to finish (join).
+        self.watcher_cancel_token.cancel();
+    }
+}
+
+impl MinimumValue {
+    pub async fn new(pgpool: PgPool) -> Self {
+        let cost_model_map: CostModelMap = Default::default();
+        let global_model: GlobalModel = Default::default();
+        Self::value_check_reload(&pgpool, cost_model_map.clone(), global_model.clone())
+            .await
+            .expect("should be able to reload cost models");
+
+        let mut pglistener = PgListener::connect_with(&pgpool.clone()).await.unwrap();
+        pglistener
+            .listen("cost_models_update_notification")
+            .await
+            .expect(
+                "should be able to subscribe to Postgres Notify events on the channel \
+                'cost_models_update_notification'",
+            );
+
+        let watcher_cancel_token = tokio_util::sync::CancellationToken::new();
+        tokio::spawn(CostModelWatcher::cost_models_watcher(
+            pgpool.clone(),
+            pglistener,
+            cost_model_map.clone(),
+            global_model.clone(),
+            watcher_cancel_token.clone(),
+        ));
+
+        Self {
+            global_model,
+            cost_model_map,
+            watcher_cancel_token,
+        }
+    }
+
+    fn get_expected_value(&self, agora_query: &AgoraQuery) -> anyhow::Result<u128> {
```

Comment:
The naming can be somewhat arbitrary. For example, here, you have a getter `get_expected_value`, but in `cost_model.rs`, you have getters named `cost_models` and `cost_model`.

Reply:
Good with either, but just pick one.

Reply:
We have `cost_models` and `cost_model` there because they map to GraphQL queries that indexer-service has.

I usually use `get_` but it's a Java mind hahaha, maybe a future refactor I should remove all getters into more meaningful names.

Reply:
I love that you thumbs-up'ed your own comment hahaha. Glad you agree with yourself :joy: